<a href="https://colab.research.google.com/github/raghadbasem80-creator/flyrank-ml-internship-raghad-/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

SyntaxError: invalid syntax (2451634716.py, line 1)

I chose Lane 2: Refresh / Content Opportunity Scoring.

I chose this lane because it helps identify which content pages should be reviewed first for refresh. Instead of checking every page manually, this approach uses search and engagement signals to prioritize pages that are more likely to benefit from updates. This can help the content team spend their time on the pages with the highest potential value.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question:
Which content pages should be reviewed first for refresh?

Decision:
Determine which pages should be prioritized for content refresh.

Who acts on it?
The content or SEO team.

Action:
Review and update the highest-priority pages before others.

Unit of Analysis:
One content page.

Output:
A ranked list of pages with a refresh priority score.

Cost of a Wrong Recommendation:
If the recommendation is wrong, the team may spend time updating pages that do not need improvement while missing pages that actually need attention. This wastes time and resources and may reduce the impact of the content refresh effort.

Why can ML help?
The dataset contains many pages and multiple measurable signals. Machine learning can identify useful patterns and support prioritization more consistently than manual review alone.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/raghadbasem80-creator/flyrank-ml-internship-raghad-.git
import os

for root, dirs, files in os.walk("flyrank-ml-internship-raghad-"):
    for file in files:
        if file == "content_refresh_anonymized.csv":
            print(os.path.join(root, file))
            import pandas as pd

df = pd.read_csv("flyrank-ml-internship-raghad-/data/raw/content_refresh_anonymized.csv")

print("Dataset shape:", df.shape)

print("\nColumns:")
print(df.columns.tolist())

print("\nSummary:")
display(df.describe(include="all"))

fatal: destination path 'flyrank-ml-internship-raghad-' already exists and is not an empty directory.
flyrank-ml-internship-raghad-/data/raw/content_refresh_anonymized.csv
Dataset shape: (30000, 44)

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']

Summary:


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
count,30000,30000,27532.000000,27532.000000,27390,27532.000000,30000,27626,22301.000000,22301.000000,...,22301,30000.000000,30000.00000,30000.000000,29875.000000,30000.000000,30000,30000,30000,26612.000000
unique,30000,32,NaN,NaN,3,NaN,3,4,NaN,NaN,...,4,NaN,NaN,NaN,NaN,NaN,4,5,5,NaN
top,content_6880eb215048,client_19581e27de,NaN,NaN,LOW,NaN,keyword article,informational,NaN,NaN,...,15000-25000,NaN,NaN,NaN,NaN,NaN,low,page_1,down,NaN
freq,1,7008,NaN,NaN,22896,NaN,27207,17235,NaN,NaN,...,12055,NaN,NaN,NaN,NaN,NaN,11248,11814,16262,NaN
mean,NaN,NaN,158.882391,0.146954,NaN,0.485342,NaN,NaN,3107.760325,20665.277835,...,NaN,0.510733,16.34238,2.534520,18.212921,0.768196,NaN,NaN,NaN,-4.785969
std,NaN,NaN,1518.270825,0.285241,NaN,2.101560,NaN,NaN,1452.382598,10115.344042,...,NaN,3.279162,15.21679,8.310096,29.472768,7.429454,NaN,NaN,NaN,473.861780
min,NaN,NaN,0.000000,0.000000,NaN,0.000000,NaN,NaN,8.000000,40.000000,...,NaN,0.000000,0.00000,0.000000,0.000000,0.000000,NaN,NaN,NaN,-100.000000
25%,NaN,NaN,0.000000,0.000000,NaN,0.000000,NaN,NaN,2413.000000,15644.000000,...,NaN,0.000000,6.20000,0.000000,0.000000,0.000000,NaN,NaN,NaN,-62.600000
50%,NaN,NaN,10.000000,0.000000,NaN,0.000000,NaN,NaN,2877.000000,19116.000000,...,NaN,0.070000,10.80000,0.000000,5.000000,0.000000,NaN,NaN,NaN,-33.500000
75%,NaN,NaN,20.000000,0.130000,NaN,0.000000,NaN,NaN,3666.000000,24011.000000,...,NaN,0.290000,22.30000,1.350000,23.530000,0.000000,NaN,NaN,NaN,0.000000


### Quick Look at the Data

The starter dataset contains **30,000 content pages** and **44 features** that describe search performance, content characteristics, engagement, and freshness.

The dataset includes **32 different clients**, making it suitable for learning patterns across multiple websites rather than a single client.

The average search volume is **158.88**, while the average page position is **16.34**. These differences suggest that pages vary considerably in performance, making content prioritization a useful machine learning problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful words: what I can and can't claim

This project provides **decision-support** by identifying observed patterns in the available data. The results can help prioritize which content pages should be reviewed first for refresh.

The findings are **observed** and **directional**, not causal. They show relationships in the dataset but do not prove that refreshing a page will improve its performance.

This project does **not** predict Google's algorithm or guarantee better search rankings. Final decisions should always include human judgment alongside the model's recommendations.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.